src/main/resources/report.csv

```diff
@@ -0,0 +1,15 @@
+fruit,quantity
+banana,152
+apple,90
+public void transaction(List<FruitTransaction> fruitTransactions) {
+        Optional.ofNullable(fruitTransactions).orElseThrow(
+                () -> new NoSuchElementException("Transaction list can`t be null"));
+        for (FruitTransaction transaction : fruitTransactions) {
+            OperationHandler operation = operationStrategy.getOperation(transaction.getOperation());
+            fruitStorageDao
+                    .add(transaction.getFruit(), operation
+                            .operate(Optional.ofNullable(fruitStorageDao
+                                            .get(transaction.getFruit())).orElse(DEFAULT_EMPTY_VALUE),
+                                    transaction.getQuantity()));
+        }
+    }
```

Comment:
is it bad?) i was trying to put optional without variable declaration, its a result, but i got it - isn`t readable

Reply:
it is .csv file)

Reply:
didn't understand then 😕

Reply:
earlier i was checking in if statement if storage does not contains key and if it was empty(null) i added 0 to balance to make an operation

Reply:
because operation method takes two parameters with representation of current balance and quantity, if we do not do that we get NPE, there is no values is storage to do operation and we need to add 0 to it, i decided to use optional for this)

Reply:
ohhhhhh
i got it ahahahaha, i was saving some code in file ahahahah my bad